crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs

```diff
@@ -132,10 +132,28 @@ impl Analyzer<'_, '_> {
             .with_child(ScopeKind::Flow, true_facts_for_rhs.clone(), |child: &mut Analyzer| -> VResult<_> {
                 child.ctx.should_store_truthy_for_access = false;
 
+                let mut type_param = vec![];
+                if let TsInstantiation(RTsInstantiation {
+                    type_args: box RTsTypeParamInstantiation { params, .. },
+                    ..
+                }) = &**right
+                {
+                    for param in params {
+                        let ty = child.validate(&**param, ());
```

Comment:
This is double validation of `right.params` so I think this is a wrong direction of change

Reply:
this code is just change type

Reply:
What is wrong?

Reply:
This calls `validate(&mut self, &RTsTypeParamInstantiation)` one more time, which is double validation

Reply:
Thanks to you, I think I found a better way.